Sources/Tabs/MainCoordinator.swift

```diff
@@ -9,141 +9,47 @@ import UIKit
-                    updateToken()
-                    windowManager.hideUnlockWindow()
-                } catch {
-                    switch error {
-                    case is JWTVerifierError,
-                        SecureStoreError.unableToRetrieveFromUserDefaults,
-                        SecureStoreError.cantInitialiseData,
-                        SecureStoreError.cantRetrieveKey:
-                        fullLogin(loginError: error)
-                    default:
-                        print("Token retrival error: \(error)")
-                    }
-                }
-            }
-        }
-    }
-    
-    @objc private func startReauth() {
-        fullLogin(loginError: TokenError.expired)
-    }
-    
-    private func fullLogin(loginError: Error? = nil) {
-        sessionManager.endCurrentSession()
-        showLogin(loginError)
-        windowManager.hideUnlockWindow()
+        addTabs()
     }
     
     func handleUniversalLink(_ url: URL) {
-        switch UniversalLinkQualifier.qualifyOneLoginUniversalLink(url) {
-        case .login:
-            loginCoordinator?.handleUniversalLink(url)
-        case .wallet:
-            if walletAvailabilityService.shouldShowFeatureOnUniversalLink {
-                addWalletTab()
-                walletCoordinator?.handleUniversalLink(url)
-            }
-        case .unknown:
-            return
-        }
-    }
-    
-    func checkAppVersion() {
-        Task {
-            do {
-                let appInfo = try await updateService.fetchAppInfo()
-                AppEnvironment.updateReleaseFlags(appInfo.releaseFlags)
-                
-                guard updateService.currentVersion >= appInfo.minimumVersion else {
-                    return // TODO: DCMAW-9866 - Present 'app update required' screen
-                }
-                
-            } catch {
-                let error = ErrorPresenter
-                    .createGenericError(errorDescription: error.localizedDescription,
-                                        analyticsService: analyticsCenter.analyticsService) {
-                        exit(0)
-                    }
-                root.present(error, animated: true)
-            }
+        if walletAvailabilityService.shouldShowFeatureOnUniversalLink {
+            addWalletTab()
+            walletCoordinator?.handleUniversalLink(url)
         }
     }
 }
 
 extension MainCoordinator {
-    private func showLogin(_ loginError: Error?) {
-        let lc = LoginCoordinator(appWindow: windowManager.appWindow,
-                                  root: UINavigationController(),
-                                  analyticsCenter: analyticsCenter,
-                                  sessionManager: sessionManager,
-                                  networkMonitor: NetworkMonitor.shared,
-                                  loginError: loginError)
-        openChildModally(lc, animated: false)
-        loginCoordinator = lc
-    }
-    
```

Comment:
There is no reference to login from the MainCoordinator class anymore as the login flow sits as a child flow to the QualifyingCoordinator. The QualifyingCoordinator should subscribe to events which may emanate from the MainCoordinator or child coordinators and present the login flow as an independent journey which depends upon state which sits across the whole app.